src/runtime/agas/server/route.cpp

```diff
@@ -75,6 +75,7 @@ namespace hpx { namespace agas { namespace server
             addr.address_ = g.lva();
         }
 
+        naming::id_type source = p.source_id();
```

Comment:
What is the rationale of this change?

Reply:
`p` is moved here: https://github.com/STEllAR-GROUP/hpx/pull/2982/files/69463c83e47825205fbb10147b2d97b6726e1242#diff-fe4bb9d1c3c7a064a0b4dd4bdae63198L87

That means that the `source_id` used later on to update the cache entry is being used from a moved from object.

Reply:
Right, thanks for catching this.